### DISCUSSION TITLE
nextjs15 react19 typescript error

### DISCUSSION THREAD

**Discussion by Tahmeed-programmer** [vercel/next.js#71619] [App Router]
i am not sure why i keep on running into this error even though i followed the nextjs docs to upgrade to 15 from 14. I wish the docs showed the proper steps dealing with react 19 typescript errors , here is my error that am facing now - TypeError: (0 , {imported module [project]/nodemodules/.pnpm/next@15.0.0_@babel+core@7.24.5_@opentelemetry+api@1.9.0react-dom@19.0.0-rc-69d4b800-20241021fdsriyefv7rxpf2d5gm2x6rvf4/nodemodules/next/dist/server/route-modules/app-page/vendored/rsc/react.js [app-rsc] (ecmascript)}.createContext) is not a function

**Comment by mhornbacher** [comment]
Did you manage to resolve this?

**Comment by stella-sre** [comment]
This error usually occurs due to version incompatibility between **React 19** RC and some dependencies.

### Try these steps:

**1. Clear the cache and reinstall:**
```bash
rm -rf node_modules .next pnpm-lock.yaml
pnpm install
```

**2. Verify that all dependencies use React 19:**
```bash
pnpm why react
```

**3. If you use external providers, update them to versions compatible with React 19.**

---

What additional dependencies do you have in your project?
